src/app/controllers/ReferralCardController.ts

```diff
@@ -0,0 +1,131 @@
+    this.app = app;
+    // ADD BINDINGS BELOW
+    this.isValid = this.isValid.bind(this); /// Include this mutation ???????
+    this.create = this.create.bind(this);
+    this.update = this.update.bind(this);
+    this.delete = this.delete.bind(this);
+  }
+
+  // ADD MUTATION METHODS BELOW
+
+  /**
+   * Create a new payout request
+   * @param {CreateReferralCardInput} referral_card - Referral Card
+   * @returns {Promise<ReferralCard>}
+   */
+  create(referral_card: CreateReferralCardInput): Promise<ReferralCard> {
+    // what goes here?
+    return new Promise((resolve, reject) => {
+      let mutationString = `
+                      mutation ($referral_card: String! ) {
+                          createReferralCard(referral_card: $referral_card) {
+                            customer_id
+                            referral_signup_code_used
+                            referrals_remaining_for_supercharged_coupon
+                          }
+                      }
+                  `;
+      this.app
+        .getAdaptor()
+        .mutate(mutationString, {
+          referral_card,
+        })
+        .then((result: MutateResult) => {
+          resolve(result.createReferralCard.id); // result does not find any Referal
+        })
+        .catch((e: any) => {
+          reject(e);
+        });
+    });
+  }
+
+  /**
+   * Update a CardReferral and return the ID of the updated object if successful
+   * @param {string} id - The id of the Referral Card to be modified
+   * @param {UpdateReferralCardInput} referral_card - The Modified Referal Card Object
+   * @returns {Promise<ReferralCard>}
+   */
+  update(
+    id: string,
+    referral_card: UpdateReferralCardInput
+  ): Promise<ReferralCard> {
+    return new Promise((resolve, reject) => {
+      let mutationString = `
+                mutation ($id: String!, $referral_card: UpdateReferralCardInput!) {
+                    updateSurvey(id: $id, referral_card: $referral_card) {
+                        _id
+                    }
+                }
+            `;
+      this.app
+        .getAdaptor()
+        .mutate(mutationString, { id, referral_card })
+        .then((result: MutateResult) => {
+          resolve(result.updateSurvey.customer_id);  // result does not find any Referal
+        })
+        .catch((e: any) => {
+          reject(e);
+        });
+    });
+  }
+
+  /**
+   * Delete a Referral Card
+   * @param {string} id - The id of the Referal Card Object
+   * @returns {Promise<string>} - Confirmation String
+   */
+  delete(id: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      let mutationString = `
+                mutation ($id: String!) {
+                  deleteReferralCard(id: $id)
+                }
+            `;
+      this.app
+        .getAdaptor()
+        .mutate(mutationString, { id })
+        .then((result: MutateResult) => {
+          resolve(result.deleteReferralCard.id); // result does not find any Referal
```

Comment:
This wont return an id

Reply:
![image](https://user-images.githubusercontent.com/7469722/109691934-cba87080-7b3c-11eb-9a6c-ada067624031.png)

@DenisMela It says it returns a string, what is the string value ??

Reply:
It returns `'Referral card has been successfully removed.'`